apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/ItemController.java

```diff
@@ -137,6 +153,15 @@ public void deleteItem(@PathVariable String appId, @PathVariable String env,
     this.itemOpenApiService.removeItem(appId, env, clusterName, namespaceName, key, operator);
   }
 
+  @PreAuthorize(value = "@consumerPermissionValidator.hasModifyNamespacePermission(#request, #appId, #namespaceName, #env)")
+  @DeleteMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/encodeItems/{key:.+}")
+  public void deleteItemByEncodeKey(@PathVariable String appId, @PathVariable String env,
+                         @PathVariable String clusterName, @PathVariable String namespaceName,
+                         @PathVariable String key, @RequestParam String operator,
+                         HttpServletRequest request) {
+    this.deleteItem(appId,env,clusterName,namespaceName,UrlUtils.decode(key),operator,request);
```

Comment:
Please use the [apollo code style](https://github.com/apolloconfig/apollo/tree/master/apollo-buildtools/style) for format the codes.
```suggestion
    this.deleteItem(appId, env, clusterName, namespaceName, UrlUtils.decode(key), operator, request);
```